test-utils/runtime/src/substrate_test_pallet.rs

```diff
@@ -0,0 +1,292 @@
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {}
+
+	// The current block number being processed. Set by `on_initialize`.
+	#[pallet::storage]
+	pub type Number<T: Config> = StorageValue<_, T::BlockNumber, OptionQuery>;
+
+	#[pallet::storage]
+	pub type NewAuthorities<T> = StorageValue<_, Vec<AuthorityId>, OptionQuery>;
+
+	#[pallet::storage]
+	pub type Authorities<T> = StorageValue<_, Vec<AuthorityId>, ValueQuery>;
+
+	#[pallet::genesis_config]
+	#[cfg_attr(feature = "std", derive(Default))]
+	pub struct GenesisConfig {
+		pub authorities: Vec<AuthorityId>,
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+			<Authorities<T>>::put(self.authorities.clone());
+		}
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(n: T::BlockNumber) -> Weight {
+			Number::<T>::put(n);
+			Weight::zero()
+		}
+
+		fn on_finalize(_n: T::BlockNumber) {}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight(100)]
+		pub fn authorities_change(
+			origin: OriginFor<T>,
+			new_authorities: Vec<AuthorityId>,
+		) -> DispatchResult {
+			frame_system::ensure_signed(origin)?;
+			<NewAuthorities<Runtime>>::put(new_authorities.to_vec());
+			Ok(())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight(100)]
+		pub fn transfer(
+			origin: OriginFor<T>,
+			transfer: Transfer,
+			_signature: Signature,
+			_exhaust_resources_when_not_first: bool,
+		) -> DispatchResult {
+			log::trace!(target: LOG_TARGET, "transfer");
+			frame_system::ensure_signed(origin)?;
+
+			//todo/cross-check: do we need to re-verify transfer (signature / nonce / balance)?
+
+			let nonce_key = transfer.from.to_keyed_vec(NONCE_OF);
+			let expected_nonce: u64 = storage::hashed::get_or(&blake2_256, &nonce_key, 0);
+			// increment nonce in storage
+			storage::hashed::put(&blake2_256, &nonce_key, &(expected_nonce + 1));
+
+			// check sender balance
+			let from_balance_key = transfer.from.to_keyed_vec(BALANCE_OF);
+			let from_balance: u64 = storage::hashed::get_or(&blake2_256, &from_balance_key, 0);
+
+			let to_balance_key = transfer.to.to_keyed_vec(BALANCE_OF);
+			let to_balance: u64 = storage::hashed::get_or(&blake2_256, &to_balance_key, 0);
+			storage::hashed::put(&blake2_256, &from_balance_key, &(from_balance - transfer.amount));
+			storage::hashed::put(&blake2_256, &to_balance_key, &(to_balance + transfer.amount));
+			Ok(())
+		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight(100)]
+		pub fn include_data(origin: OriginFor<T>, _data: Vec<u8>) -> DispatchResult {
+			log::trace!(target: LOG_TARGET, "include_data");
+			frame_system::ensure_signed(origin)?;
+			Ok(())
+		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight(100)]
+		pub fn storage_change_unsigned(
```

Comment:
This can be removed as `storage_change`. Not sure we really need an unsigned version? 

Generally speaking, you could already had merged both method and just dropped the `ensure_signed` check.

Reply:
I chose to left `unsigned` version to enforce test author to explicitly choose `unsigned` if he wants to use unsigned Extrinsics. We can check for this exact call in `ValidateUnsigned` for `substrate-test-pallet`, and this allows us to avoid unintended usages of unsigned Extrinsics. (I am quite defensive in my reasoning here).

I'd prefer to stay explicit, even if it seems redundant in naming and functionality.